src/erc4626/ERC4626Oracle.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {IERC20} from "./IERC20.sol";
+import {IERC4626} from "./IERC4626.sol";
+
+contract ERC4626Oracle {
+    error OracleUnsupportedPair();
+
+    IERC4626 public immutable VAULT;
+    address public immutable ASSET;
+
+    constructor(address _vault) {
+        VAULT = IERC4626(_vault);
+        ASSET = VAULT.asset();
+    }
+
+    function valueOf(address base, address quote, uint256 baseAmount) external view returns (uint256 quoteAmount) {
+        if (quote == address(VAULT) && base == ASSET) {
```

Comment:
```suggestion
        if (base == ASSET && quote == address(VAULT)) {
```
Same effect, but it is a good practice to always put the base before the quote when possible, as in the function arguments.